### PR TITLE
270-incorrect-XML-marshalling-in-Xliff2.0

### PR DESCRIPTION
**Pull request by DarjaLeonova**

### CODE REVIEW

File: pkg/convert/xliff2_test.go [hunk 1]
```diff
@@ -19,55 +19,57 @@ import (
...
-	b.WriteString("<file>")
-
-	writeMsg := func(s string) { fmt.Fprintf(b, "<segment><target>%s</target></segment>", s) }
 	if translation.Original {
-		writeMsg = func(s string) { fmt.Fprintf(b, "<segment><source>%s</source></segment>", s) }
+		xliff.SrcLang = translation.Language
+		xliff.TrgLang = language.Und
+	} else {
+		xliff.SrcLang = language.Und
+		xliff.TrgLang = translation.Language
```
Comment (VladislavsPerkanuks) [inline pkg/convert/xliff2_test.go:31]:
```suggestion
		xliff.SrcLang = translation.Language
	} else {
		xliff.TrgLang = translation.Language
```

File: pkg/convert/xliff2_test.go [hunk 2]
```diff
@@ -19,55 +19,57 @@ import (
...
 	}
 
 	for _, msg := range translation.Messages {
-		fmt.Fprintf(b, "<unit id=\"%s\">", msg.ID)
+		xmlMsg := unit{
+			ID:     msg.ID,
+			Source: "",
+			Target: "",
```
Comment (VladislavsPerkanuks) [inline pkg/convert/xliff2_test.go:38]:
```suggestion
			ID:     msg.ID,
```

File: pkg/convert/xliff2_test.go [hunk 3]
```diff
@@ -19,55 +19,57 @@ import (
...
-			fmt.Fprintf(b, "</notes>")
+			xmlMsg.Notes = &notes
 		}
 
-		writeMsg(msg.Message)
+		xliff.File.Units = append(xliff.File.Units, xmlMsg)
+	}
 
-		b.WriteString("</unit>")
+	xmlData, err := xml.MarshalIndent(xliff, "", "  ")
```
Comment (VladislavsPerkanuks) [inline pkg/convert/xliff2_test.go:64]:
```suggestion
	xmlData, err := xml.Marshal(xliff)
```

File: pkg/convert/xliff2_test.go [hunk 4]
```diff
@@ -19,55 +19,57 @@ import (
...
+		fmt.Printf("marshaling XLIFF2.0: %v\n", err)
+		return nil
 	}
 
-	b.WriteString("</file>")
-	b.WriteString("</xliff>")
+	xmlWithDeclaration := []byte(xml.Header + string(xmlData))
 
-	return b.Bytes()
+	return xmlWithDeclaration
```
Comment (VladislavsPerkanuks) [inline pkg/convert/xliff2_test.go:72]:
```suggestion
	return append([]byte(xml.Header), xmlData...)
```

**Review by kaaposc — APPROVED** [review]

**Review by VladislavsPerkanuks — APPROVED** [review]
looks fine, so xliff 1.2, in another PR ? 🤔

**Pull request merged**